lib/services/access_list.go

```diff
@@ -282,6 +286,44 @@ func UnmarshalAccessListReview(data []byte, opts ...MarshalOption) (*accesslist.
 	return &review, nil
 }
 
+// ParseAccessListNextKey creates a pagination key for access list queries based on the provided
+// next token string and index name.
+//
+// The nextTokenString is expected to be a slash-separated string with the following format:
+//   - Field 0: accessList.GetName()
+//   - Field 1: accessList.Spec.Audit.NextAuditDate.Format(time.DateOnly)
+//
+// Supported index names:
+//   - "name": Returns the access list name (field 0)
+//   - "auditNextDate": Returns the audit date followed by the name (field 1 + "/" + field 0)
+//
+// Returns an error if the nextTokenString has fewer than 2 fields or if an unsupported
+// index name is provided. An empty nextTokenString will return an empty string and no error.
+func ParseAccessListNextKey(nextTokenString string, indexName string) (string, error) {
+	if nextTokenString == "" {
+		return "", nil
+	}
+	fields := strings.Split(nextTokenString, "/")
+	if len(fields) < 2 {
+		return "", trace.BadParameter("invalid nextToken supplied")
+	}
```

Comment:
Prefer using `strings.Cut` and checking for the "ok" return bool rather than splitting into a slice.